NuKeeper.Tests/RepositoryInspection/PackageUpdateSetTests.cs

```diff
@@ -0,0 +1,122 @@
+using System.Collections.Generic;
+using System.Linq;
+using NuGet.Packaging.Core;
+using NuGet.Versioning;
+using NuKeeper.RepositoryInspection;
+using NUnit.Framework;
+
+namespace NuKeeper.Tests.RepositoryInspection
+{
+    [TestFixture]
+    public class PackageUpdateSetTests
+    {
+        [Test]
+        public void NullNewPackageId_IsNnotAllowed()
+        {
+            var packages = new List<PackageInProject>
+            {
+                new PackageInProject("foo", "1.0.0", PathToProjectOne())
+            };
+
+            Assert.Throws<ArgumentNullException>(() => new PackageUpdateSet(null, packages));
+        }
+
+        [Test]
+        public void NullPackages_IsNotAllowed()
+        {
+            Assert.Throws<ArgumentNullException>(() => new PackageUpdateSet(NewPackageId(), null));
+        }
+        [Test]
+        public void EmptyPackages_IsNotAllowed()
+        {
+            Assert.Throws<ArgumentException>(
+                () => new PackageUpdateSet(NewPackageId(), Enumerable.Empty<PackageInProject>()));
+        }
+
+        [Test]
+        public void OneUpdate_IsValid()
+        {
+            var newPackage = NewPackageId();
+
+            var currentPackages = new List<PackageInProject>
+            {
+                new PackageInProject("foo", "1.0.0", PathToProjectOne())
+            };
+
+            var updates = new PackageUpdateSet(newPackage, currentPackages);
+
+            Assert.That(updates, Is.Not.Null);
+            Assert.That(updates.NewPackage, Is.EqualTo(NewPackageId()));
+            Assert.That(updates.CurrentPackages, Is.Not.Null);
+            Assert.That(updates.CurrentPackages.Count, Is.EqualTo(1));
+
+            Assert.That(updates.PackageId, Is.EqualTo("foo"));
+            Assert.That(updates.NewVersion, Is.EqualTo(newPackage.Version));
+        }
+
+        [Test]
+        public void TwoUpdates_AreValid()
+        {
+            var newPackage = NewPackageId();
+
+            var currentPackages = new List<PackageInProject>
+            {
+                new PackageInProject("foo", "1.0.0", PathToProjectOne()),
+                new PackageInProject("foo", "1.0.1", PathToProjectTwo())
+            };
+
+            var updates = new PackageUpdateSet(newPackage, currentPackages);
+
+            Assert.That(updates, Is.Not.Null);
+            Assert.That(updates.NewPackage, Is.EqualTo(NewPackageId()));
+            Assert.That(updates.CurrentPackages, Is.Not.Null);
+            Assert.That(updates.CurrentPackages.Count, Is.EqualTo(2));
+
+            Assert.That(updates.PackageId, Is.EqualTo("foo"));
+            Assert.That(updates.NewVersion, Is.EqualTo(newPackage.Version));
+        }
+
+        [Test]
+        public void CannotHaveUpdateForDifferentPackageToNewVersion()
+        {
+            var newPackage = NewPackageId();
+
+            var currentPackages = new List<PackageInProject>
+            {
+                new PackageInProject("bar", "1.0.0", PathToProjectOne())
```

Comment:
the code for this test doesn't explicitly declare that "bar" is different to "foo" - have to f12 around for that. Is there a way of making the code of this test more declarative?

e.g. `NewPackageId()` -> `NewPackageWithId("foo")`